app/models/account.rb

```diff
@@ -1,9 +1,19 @@
 # Customer organization account
 class Account < ActiveRecord::Base
+  # @param [String] piece the tenant piece of the canoncial name
+  # @return [String] full canonical name
+  # @see Settings.multitenancy.default_host
+  def self.default_cname(piece)
+    return unless piece
+    default_host = Settings.multitenancy.default_host
+    format(default_host, tenant: piece.parameterize)
+  end
+
   attr_readonly :tenant
-  validates :name, presence: true
+  # name is unused after create, only used by sign_up/new forms
+  validates :name, presence: true, unless: 'cname.present? && cname != default_cname("")'
```

Comment:
Yes, that is acceptable.  One of the fields (`cname`) obtains a default value based on another field (`name`) passed through a configurable template string.  Presently, it may have already obtained that value, but erroneously (`name` was empty).  Calling `default_cname('')`, validators know what to exclude as effectively a `nil` value.